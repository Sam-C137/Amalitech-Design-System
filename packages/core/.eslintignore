dist
loader
www
**/*.e2e.ts
**/*.spec.tsx
**/*.spec.ts
