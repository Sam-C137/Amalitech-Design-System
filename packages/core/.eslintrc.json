{
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "extends": [
    "plugin:@stencil-community/recommended"
  ],
  "ignorePatterns": [
    "**/*.stories.tsx",
    "**/*.spec.*",
    "**/*.e2e.*"
  ],
  "rules": {
    "react/jsx-no-bind": "off"
  }
}
