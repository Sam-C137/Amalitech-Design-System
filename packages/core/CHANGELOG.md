# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.8.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.2...v0.8.3) (2024-07-18)


### Bug Fixes

* min max value attribute changes ([8276d9d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/8276d9dd0d176071116497f01c12c92caecd5706))





## [0.8.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.1...v0.8.2) (2024-06-02)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.8.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.0...v0.8.1) (2024-06-01)


### Bug Fixes

* tests ([05989c4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/05989c4eaa674a902264d87c3044b6eb91794698))





# [0.8.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.7.0...v0.8.0) (2024-05-27)


### Features

* add double border to radio ([21d478d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/21d478d51df6c38aa88b8a1b429b76baa7371040))





# [0.7.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.2...v0.7.0) (2024-05-14)


### Features

* add clear method to select ([689be3d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/689be3df7d4c3f895ef8b2f11fc8bc24a2f78ab3))
* create select value accessor directive ([d5957c8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/d5957c85d858f02ec8506671ae7ac142421a34c5))





## [0.6.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.1...v0.6.2) (2024-05-06)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.6.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.0...v0.6.1) (2024-05-06)


### Bug Fixes

* removed default switch margin ([503f85d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/503f85da2f380b6b7a89e3a46015ea367d67a2ce))





# [0.6.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.5.0...v0.6.0) (2024-04-30)


### Bug Fixes

* pagination spacing ([00c99e4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/00c99e4e17edf1f4d145135d8f1da444f6cb6ed6))


### Features

* exposed totalPagesCount in pagination ([4105737](https://github.com/Amali-Tech/Amalitech-Design-System/commit/4105737d72ba70a4bc0b9a7fa67dca5e73c2bc5b))





# [0.5.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.4.0...v0.5.0) (2024-04-28)


### Features

* utilities align self ([ab106b6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/ab106b6f3752e20155c9b307d89cbcc4b12c79ec))





# [0.4.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.7...v0.4.0) (2024-04-28)


### Features

* add z-index to utilities ([75efd78](https://github.com/Amali-Tech/Amalitech-Design-System/commit/75efd78fbf0a22ce34139c7569a5ee894faaa088))





## [0.3.7](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.6...v0.3.7) (2024-04-27)


### Bug Fixes

* date picker open method: ([710e97b](https://github.com/Amali-Tech/Amalitech-Design-System/commit/710e97b29a3538231ada47ffddd7314be1766559))





## [0.3.6](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.5...v0.3.6) (2024-04-24)


### Bug Fixes

* centered the checkbox in it's label ([e7fbbaa](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e7fbbaaf3f79a6d0185777a0ce52a44a6b6f84e8))






## [0.3.5](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.4...v0.3.5) (2024-04-15)


### Bug Fixes

* build ([12140c9](https://github.com/Amali-Tech/Amalitech-Design-System/commit/12140c95e8be561d9f26d67c15b71c2beb50dee0))
* build ([a38dd2e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a38dd2e615bedc32ab7a68ce841a9b04e4f8267b))
* card tests ([5c99406](https://github.com/Amali-Tech/Amalitech-Design-System/commit/5c99406dc9796839ee06326fdaf2b8471e2d554d))
* starting with a value ([fb24ce3](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fb24ce3cf7510dc5cca9c868aa03b48e3aa8acdb))
* tests ([373434e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/373434e70de883671a1f828d1cbda137d717978a))
* tests ([eb11522](https://github.com/Amali-Tech/Amalitech-Design-System/commit/eb11522d23bf5b31c41228ba9f522998641c677f))
* tests ([e44f72d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e44f72d42a5b9fc587edfbeecafd3c1dc89b799a))
* tests ([799a98c](https://github.com/Amali-Tech/Amalitech-Design-System/commit/799a98cf03b6c008f749fdb754187abcca9cface))
* tests ([fcb7f47](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fcb7f47051af324d7d5ea1c0a285f5f2645788be))
* tests ([66ea8f8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/66ea8f888460610f550d88b51f1d4ec870444072))






## [0.3.4](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.3...v0.3.4) (2024-04-05)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.2...v0.3.3) (2024-04-04)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.1...v0.3.2) (2024-04-03)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.3.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.0...v0.3.1) (2024-04-03)


### Bug Fixes

* tests ([a1ca9e2](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a1ca9e26ce368e9d26b114605ad3869ccaa5dd8d))






# [0.3.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.2...v0.3.0) (2024-04-02)


### Features

* form submission for input & button ([2cc64d6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/2cc64d69378de5a3ec3e7c9da20f53e36e3a5dcc))






## [0.2.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.1...v0.2.2) (2024-03-28)

**Note:** Version bump only for package @amalitech-design-system/core
