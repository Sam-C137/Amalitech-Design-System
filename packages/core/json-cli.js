const fs = require('fs').promises;
const path = require('path');

const svgFolderPath = './src/assets/icons/linear';
const outputFilePath = './src/assets/icons/linear-icons.json';

async function main() {
  const svgs = await getSvgFiles(svgFolderPath);
  const data = await Promise.all(svgs.map(parseSvgFile));
  await writeJsonToFile(outputFilePath, data);
  console.log(`Generated JSON file: ${outputFilePath}`);
}

async function getSvgFiles(folderPath) {
  const files = await fs.readdir(folderPath);
  return files.filter(file => path.extname(file) === '.svg');
}

async function parseSvgFile(fileName) {
  const filePath = path.join(svgFolderPath, fileName);
  const svgContent = await fs.readFile(filePath, 'utf-8');
  const label = path.basename(fileName, '.svg');

  return { label, icon: svgContent };
}

async function writeJsonToFile(filePath, data) {
  const jsonData = JSON.stringify(data, null, 2);
  await fs.writeFile(filePath, jsonData);
}

main().catch(error => {
  console.error(error);
});
