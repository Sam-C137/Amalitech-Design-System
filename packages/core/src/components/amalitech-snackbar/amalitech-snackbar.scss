:host {
  /**
    * @prop --background-color: Background color of the alert
    * @prop --color: Text color of the alert
    *
    * @prop --padding-top: Padding top of the alert
    * @prop --padding-right: Padding right of the alert
    * @prop --padding-bottom: Padding bottom of the alert
    * @prop --padding-left: Padding left of the alert
    *
    * @prop --font-family: Font family of the alert text
    * @prop --font-size: Font size of the alert text
    * @prop --font-weight: Font weight of the alert text
    */
  -webkit-font-smoothing: antialiased;
  padding: var(--padding-top, 0.5rem) var(--padding-right, 1rem) var(--padding-bottom, 0.5rem) var(--padding-left, 1rem);
  min-height: var(--height, 1.5rem);
  color: var(--color, $alert-info-color);
  background-color: var(--background-color, $light-400);
  border: 0.0625rem solid $neutral-100;
  font-family: var(--font-family, "Work Sans", Helvetica, Arial, sans-serif);
  font-size: var(--font-size, 0.8125rem);
  font-weight: var(--font-weight, normal);
  line-height: 1;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  .snackbar {
    &__content {
      display: block;
      height: 100%;
      width: 100%;
      box-sizing: border-box;
    }
    &__content .icon{
      display: flex;
      align-items: center;
    }
    &__content .section{
      display: flex;
      gap: 0.5rem;
    }
    &__content .cta{
      color: $primary;
      
    }
    &__content .cta > a{
      text-decoration: none;
      color: $primary;
      &:hover {
        cursor: pointer;
        color: $orange-300;
      }
    }
    &__close-text {
      cursor: pointer;
      opacity: .5;
      position: absolute;
      top: 0;
      right: 0;
      font-size: 1.35rem;
      padding: .55rem 0.8rem;
      text-shadow: 0 0.0625rem 0 $light;
      &:hover {
        opacity: .85;
      }
    }
    &__dismissible {
      padding-right: 1.25rem;
    }
  }
}

:host(.rounded) {
  border-radius: var(--border-radius, 0.5rem);
}
