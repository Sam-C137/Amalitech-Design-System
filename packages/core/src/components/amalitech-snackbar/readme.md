# amalitech-design-system



<!-- Auto Generated Below -->


## Properties

| Property           | Attribute             | Description                                                       | Type      | Default    |
| ------------------ | --------------------- | ----------------------------------------------------------------- | --------- | ---------- |
| `autoClose`        | `auto-close`          | Auto Close Enables the auto close functionality                   | `boolean` | `false`    |
| `callToActionText` | `call-to-action-text` | Call To Action Text Sets the call-to-action text                  | `string`  | `''`       |
| `closeSecs`        | `close-secs`          | Sets the duration before the snackbar closes                      | `number`  | `10`       |
| `closeable`        | `closeable`           | Closeable Shows a close button if true                            | `boolean` | `false`    |
| `href`             | `href`                | href attribute Sets the url of the snackbar                       | `string`  | `''`       |
| `rounded`          | `rounded`             | Rounded Sets the border radius of the snackbar to round or square | `boolean` | `false`    |
| `snackbarText`     | `snackbar-text`       | Snackbar text Sets the Snackbar text                              | `string`  | `''`       |
| `target`           | `target`              | target attribute specifies how to open the url of the snackbar    | `string`  | `'_blank'` |


## Events

| Event    | Description                                            | Type               |
| -------- | ------------------------------------------------------ | ------------------ |
| `closed` | This event is emitted when the close button is clicked | `CustomEvent<any>` |


## CSS Custom Properties

| Name                 | Description                   |
| -------------------- | ----------------------------- |
| `--background-color` | Background color of the alert |
| `--color`            | Text color of the alert       |
| `--font-family`      | Font family of the alert text |
| `--font-size`        | Font size of the alert text   |
| `--font-weight`      | Font weight of the alert text |
| `--padding-bottom`   | Padding bottom of the alert   |
| `--padding-left`     | Padding left of the alert     |
| `--padding-right`    | Padding right of the alert    |
| `--padding-top`      | Padding top of the alert      |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
