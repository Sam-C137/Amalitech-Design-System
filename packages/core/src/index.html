<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Stencil Component Starter</title>
    <script type="module" src="/build/amalitech-design-system.esm.js"></script>
    <script nomodule src="/build/amalitech-design-system.js"></script>
    <link rel="stylesheet" href="/build/amalitech-design-system.css" />
    <style>
      h1 {
        text-align: center;
        font-style: normal;
        line-height: 2.5rem;
        font-family: var(--amalitech-font-variant-body-family);
        color: var(--amalitech-color-neutral-700);
        font-size: var(--amalitech-font-variant-heading-3-size);
        font-weight: var(--amalitech-font-weight-bold);
        margin: 0;
        padding: 0;
      }
    </style>
  </head>

  <body>
    <h1>This is ADS</h1>
  </body>
</html>
