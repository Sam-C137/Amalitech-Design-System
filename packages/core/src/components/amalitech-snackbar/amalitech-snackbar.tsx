import { Component, Host, h, Prop, State, Element, Event, EventEmitter } from '@stencil/core';
import { SnackbarIcons } from '../../utils/svg-icons';

@Component({
  tag: 'amalitech-snackbar',
  styleUrl: 'amalitech-snackbar.scss',
  shadow: true,
})
export class AmalitechSnackbar {
  /**
   * Auto Close
   * Enables the auto close functionality
   */
  @Prop() autoClose: boolean = false;

  /**
   * Sets the duration before the snackbar closes
   */
  @Prop() closeSecs: number = 10;

  /**
   * Closeable
   * Shows a close button if true
   */
  @Prop() closeable: boolean = false;

  /**
   * Rounded
   * Sets the border radius of the snackbar to round or square
   */
  @Prop() rounded: boolean = false;

  /**
   * Snackbar text
   * Sets the Snackbar text
   */
  @Prop({ reflect: true }) snackbarText: string = '';

  /**
   * Call To Action Text
   * Sets the call-to-action text
   */
  @Prop({ reflect: true }) callToActionText: string = '';

  /**
   * href attribute
   * Sets the url of the snackbar
   */
  @Prop() href: string = '';

  /**
   * target attribute
   * specifies how to open the url of the snackbar
   */
  @Prop() target: string = '_blank';

  @State() countDownTimerId: number | undefined = undefined;

  @Element() el: HTMLAmalitechSnackbarElement;

  private close = () => {
    this.closed.emit();
    this.el.remove();
  };

  /**
   * This event is emitted when the close button is clicked
   */
  @Event() closed: EventEmitter;

  private renderCloseText = () => {
    if (this.closeable) {
      return (
        <span class="snackbar__close-text" onClick={this.close}>
          {SnackbarIcons.closeIcon()}
        </span>
      );
    }
    return null;
  };

  componentWillLoad() {
    if (this.el.hasAttribute('auto-close')) {
      this.countDownTimerId = window.setTimeout(() => {
        this.close();
      }, this.closeSecs * 1000);
    }
  }

  disconnectedCallback() {
    if (this.autoClose) {
      window.clearInterval(this.countDownTimerId);
    }
  }

  render() {
    return (
      <Host
        class={{
          [`rounded`]: this.rounded,
        }}
      >
        <div
          class={{
            snackbar__content: true,
            snackbar__dismissible: this.closeable,
          }}
        >
          <div class={'section'}>
            <div class={'icon'}>{SnackbarIcons.errorIcon()}</div>
            <p>{this.snackbarText}</p>
            <p class={'cta'}>
              {this.callToActionText !== undefined && this.href !== undefined ? (
                <a href={this.href} target={this.target}>
                  {this.callToActionText}
                </a>
              ) : (
                this.callToActionText
              )}
            </p>
          </div>
        </div>
        {this.renderCloseText()}
      </Host>
    );
  }
}
