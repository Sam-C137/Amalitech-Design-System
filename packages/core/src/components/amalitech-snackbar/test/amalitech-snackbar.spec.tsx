import { newSpecPage } from '@stencil/core/testing';
import { AmalitechSnackbar } from '../amalitech-snackbar';

jest.setTimeout(30000);

describe('amalitech-snackbar', () => {
  let page;
  beforeEach(async () => {
    page = await newSpecPage({
      components: [AmalitechSnackbar],
      html: `<amalitech-snackbar></amalitech-snackbar>`,
    });
  });

  it('should render with default props', async () => {
    expect(page.root).toMatchSnapshot();
  });

  it('should render when closeable prop is provided', async () => {
    page.root.closeable = true;
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('should render when callToAction and href props are provided', async () => {
    page.root.callToAction = 'Click me';
    page.root.href = 'https://example.com';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
});
