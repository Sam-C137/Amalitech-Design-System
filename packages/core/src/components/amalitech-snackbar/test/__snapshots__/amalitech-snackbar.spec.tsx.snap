// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amalitech-snackbar should render when callToAction and href props are provided 1`] = `
<amalitech-snackbar call-to-action-text="" snackbar-text="">
  <mock:shadow-root>
    <div class="snackbar__content">
      <div class="section">
        <div class="icon">
          <svg height="16" viewBox="0 0 512 512" width="16" xmlns="http://www.w3.org/2000/svg">
            <path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24v112c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zm-32 224a32 32 0 1 1 64 0 32 32 0 1 1-64 0z" fill="#D14343"></path>
          </svg>
        </div>
        <p></p>
        <p class="cta">
          <a href="https://example.com" target="_blank"></a>
        </p>
      </div>
    </div>
  </mock:shadow-root>
</amalitech-snackbar>
`;

exports[`amalitech-snackbar should render when closeable prop is provided 1`] = `
<amalitech-snackbar call-to-action-text="" snackbar-text="">
  <mock:shadow-root>
    <div class="snackbar__content snackbar__dismissible">
      <div class="section">
        <div class="icon">
          <svg height="16" viewBox="0 0 512 512" width="16" xmlns="http://www.w3.org/2000/svg">
            <path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24v112c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zm-32 224a32 32 0 1 1 64 0 32 32 0 1 1-64 0z" fill="#D14343"></path>
          </svg>
        </div>
        <p></p>
        <p class="cta">
          <a href="" target="_blank"></a>
        </p>
      </div>
    </div>
    <span class="snackbar__close-text">
      <svg height="16" viewBox="0 0 512 512" width="16" xmlns="http://www.w3.org/2000/svg">
        <path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm-81-337c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z" fill="#696F8C"></path>
      </svg>
    </span>
  </mock:shadow-root>
</amalitech-snackbar>
`;

exports[`amalitech-snackbar should render with default props 1`] = `
<amalitech-snackbar call-to-action-text="" snackbar-text="">
  <mock:shadow-root>
    <div class="snackbar__content">
      <div class="section">
        <div class="icon">
          <svg height="16" viewBox="0 0 512 512" width="16" xmlns="http://www.w3.org/2000/svg">
            <path d="M256 512a256 256 0 1 0 0-512 256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24v112c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zm-32 224a32 32 0 1 1 64 0 32 32 0 1 1-64 0z" fill="#D14343"></path>
          </svg>
        </div>
        <p></p>
        <p class="cta">
          <a href="" target="_blank"></a>
        </p>
      </div>
    </div>
  </mock:shadow-root>
</amalitech-snackbar>
`;
